### PR TITLE
feat: add support transparency

### PR DESCRIPTION
**Pull request by mbfakourii**
Add support transparency export (thanks @brendan-duncan in https://github.com/brendan-duncan/image/issues/663).
Add support export base maximum frame size (thanks @brendan-duncan in https://github.com/brendan-duncan/image/issues/664).
Improve example project base last flutter changes.

Closes: #27
Closes: #39

## Pre-launch Checklist

- [x] I made sure the project builds.
- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is does not need version changes.
- [x] I updated `CHANGELOG.md` to add a description of the change.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I rebased onto `main`.
- [ ] I added new tests to check the change I am making, or this PR does not need tests.
- [ ] I made sure all existing and new tests are passing.
- [x] I ran `dart format .` and committed any changes.
- [x] I ran `flutter analyze` and fixed any errors.

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -14,7 +14,7 @@ environment:
 dependencies:
   flutter:
     sdk: flutter
-  image: ^3.0.2
```
Comment (ueman) [inline pubspec.yaml:17]: Does it require the newer `image` dependency? If not, you can just set `>=3.0.2 < 5.0.0` and allow people to use a wider range of image dependencies.

File: lib/src/exporter.dart [hunk 2]
```diff
@@ -55,10 +55,48 @@ class Exporter {
...
-      decodedImage.duration = frame.durationInMillis;
-      animation.addFrame(decodedImage);
+      decodedImage.frameDuration = frame.durationInMillis;
+      mainImage.frames.add(encodeGifWIthTransparency(decodedImage));
+    }
+
+    return image.encodeGif(mainImage);
+  }
+
+  static image.PaletteUint8 convertPalette(image.Palette palette) {
```
Comment (ueman) [inline lib/src/exporter.dart:65]: This should probably be a private method.

File: lib/src/exporter.dart [hunk 3]
```diff
@@ -55,10 +55,48 @@ class Exporter {
...
+    final newPalette = image.PaletteUint8(palette.numColors, 4);
+    for (var i = 0; i < palette.numColors; i++) {
+      newPalette.setRgba(
+          i, palette.getRed(i), palette.getGreen(i), palette.getBlue(i), 255);
     }
-    return image.encodeGifAnimation(animation);
+    return newPalette;
+  }
+
+  static image.Image encodeGifWIthTransparency(image.Image srcImage,
```
Comment (ueman) [inline lib/src/exporter.dart:74]: This should probably be a private method.

**Review by ueman — COMMENTED** [review]
This is awesome, thank you!
Please also increase the version in the pubspec and add an entry to the changelog.md file.

File: pubspec.yaml (hunk 1 above)
Comment (mbfakourii) [reply pubspec.yaml:17]: Yes, version 4 is needed

File: lib/src/exporter.dart (hunk 2 above)
Comment (mbfakourii) [reply lib/src/exporter.dart]: Done

File: lib/src/exporter.dart (hunk 3 above)
Comment (mbfakourii) [reply lib/src/exporter.dart]: Done

**Review by ueman — APPROVED** [review]

**Comment by ueman** [comment]
I can't release a new version, so that has to be done by Baseflow.

cc @mvanbeusekom @JeroenWeener @TimHoogstrate

**Comment by martijn00** [comment]
@ueman I'll rework the CI so it supports pushing new releases from there.

**Comment by martijn00** [comment]
@ueman I've implemented it, fixed some issues and released a new version. You can now make a new release by just setting a new tag. You need to make sure the changelog and version is updated before doing that.

**Pull request merged**